satpy/readers/modis_l2.py

```diff
@@ -145,6 +145,45 @@ def get_dataset(self, dataset_id, dataset_info):
         self._add_satpy_metadata(dataset_id, dataset)
         return dataset
 
+    def available_datasets(self, configured_datasets):
+        """Add dataset information from arbitrary level 2 files.
+
+        Adds dataset information not specifically specified in reader yaml file
+        from arbitrary modis level 2 product files to available datasets.
+
+        Notes:
+             Currently only adds 2D datasets and does not decode bit encoded information.
+        """
+        # pass along yaml configured (handled) datasets and collect their file keys to check against dynamically
+        # collected variables later on.
+        handled = set()
+        for is_avail, ds_info in (configured_datasets or []):
+            file_key = ds_info.get("file_key", ds_info["name"])
+            handled.add(file_key)
+
+            if is_avail is not None:
+                yield is_avail, ds_info
+                continue
+            yield self.file_type_matches(ds_info["file_type"]), ds_info
+
+        res_dict = {5416: 250, 2708: 500, 1354: 1000, 270: 5000, 135: 10000}
+
+        # get variables from file dynamically and only add those which are not already configured in yaml
+        for var_name, val in self.sd.datasets().items():
+            if var_name in handled:
+                continue
+            if len(val[0]) == 2:
+                resolution = res_dict.get(val[1][-1])
+                if resolution is not None:
+                    ds_info = {
+                        "file_type": self.filetype_info["file_type"],
+                        "resolution": resolution,
+                        "name": var_name,
+                        "file_key": var_name,
+                        "coordinates": ["longitude", "latitude"]
+                    }
+                    yield True, ds_info
```

Comment:
Maybe this would make CodeScene happier?
```suggestion
            if len(val[0]) != 2:
                continue
            resolution = res_dict.get(val[1][-1])
            if resolution is not None:
                ds_info = {
                    "file_type": self.filetype_info["file_type"],
                    "resolution": resolution,
                    "name": var_name,
                    "file_key": var_name,
                    "coordinates": ["longitude", "latitude"]
                }
                yield True, ds_info
```

Reply:
Let's find out!